idea-plugin/src/main/kotlin/arrow/meta/ide/plugins/proofs/inspections/implicitcoercion.kt

```diff
@@ -0,0 +1,58 @@
+package arrow.meta.ide.plugins.proofs.inspections
+
+import arrow.meta.ide.IdeMetaPlugin
+import arrow.meta.ide.dsl.IdeSyntax
+import arrow.meta.ide.plugins.proofs.inspections.PairTypes.Companion.pairOrNull
+import arrow.meta.ide.plugins.proofs.markers.coercionMessage
+import arrow.meta.phases.CompilerContext
+import arrow.meta.phases.ExtensionPhase
+import arrow.meta.plugins.proofs.phases.areTypesCoerced
+import arrow.meta.plugins.proofs.phases.coerceProof
+import com.intellij.codeHighlighting.HighlightDisplayLevel
+import com.intellij.codeInspection.ProblemHighlightType
+import org.jetbrains.kotlin.idea.inspections.AbstractApplicabilityBasedInspection
+import org.jetbrains.kotlin.psi.KtDotQualifiedExpression
+
+/**
+ * [implicitCoercionInspection]: for explicit coercion to make it implicit
+ */
+val IdeMetaPlugin.implicitCoercionInspection: ExtensionPhase
+  get() = addLocalInspection(
+    inspection = implicitCoercionInspectionSyntax,
+    level = HighlightDisplayLevel.WARNING,
```

Comment:
Is there a reason, why this is a warning and the explicit is an info?

Reply:
Information doesn't highlight (as we want it):
- Enforces {@link com.intellij.codeHighlighting.HighlightDisplayLevel#DO_NOT_SHOW} severity level. Please ensure that if used from inspection explicitly, corresponding problem is added in {@code onTheFly} mode only.
Warning does highlight:
- We want to make the user to go for less verbosity

Reply:
Perfect. Thanks for clarifying